routers/repo/migrate.go

```diff
@@ -188,3 +192,29 @@ func MigratePost(ctx *context.Context, form auth.MigrateRepoForm) {
 
 	handleMigrateError(ctx, ctxUser, err, "MigratePost", tpl, &form)
 }
+
+// CancelMigration cancels a running migration
+func CancelMigration(ctx *context.Context) {
+	if !ctx.Repo.IsOwner() {
+		ctx.Error(http.StatusNotFound)
+		return
+	}
+
+	if ctx.Repo.Repository.Status != models.RepositoryBeingMigrated {
+		ctx.Error(http.StatusConflict, "repo already migrated")
+		return
+	}
+
+	for _, ps := range process.GetManager().Processes() {
+		if ps.Description == fmt.Sprintf("MigrateTask: %s", ctx.Repo.Repository.FullName()) {
+			log.Trace("Migration Canceled: %s", ctx.Repo.Repository.FullName())
+			process.GetManager().Cancel(ps.PID)
+			ctx.Flash.Success(ctx.Tr("repo.migrate.cancelled"))
+			ctx.Redirect(ctx.Repo.Owner.DashboardLink())
+			return
+		}
+	}
```

Comment:
It'd be better to store the pid on the task table and grab it from there.

Reply:
... In the end we need UUIDs